napari/layers/_labels_layer/model.py

```diff
@@ -0,0 +1,226 @@
+        super().__init__(visual, name)
+        self.events.add(colormap=Event, interpolation=Event)
+
+        # update flags
+        self._need_display_update = False
+        self._need_visual_update = False
+
+        self._qt_properties = QtImageLayer(self)
+        self._qt_controls = QtImageControls(self)
+
+    def new_colormap(self):
+        seed = np.random.random((self._image.ndim,))
+        self.colormap = colormaps.label_colormap(self._image, seed=seed)
+
+    @property
+    def image(self):
+        """np.ndarray: Image data.
+        """
+        return self._image
+
+    @image.setter
+    def image(self, image):
+        self._image = image
+        self.refresh()
+
+    @property
+    def meta(self):
+        """dict: Image metadata.
+        """
+        return self._meta
+
+    @meta.setter
+    def meta(self, meta):
+        self._meta = meta
+        self.refresh()
+
+    @property
+    def data(self):
+        """tuple of np.ndarray, dict: Image data and metadata.
+        """
+        return self.image, self.meta
+
+    @data.setter
+    def data(self, data):
+        self._image, self._meta = data
+        self.refresh()
+
+    def _get_shape(self):
+        return self.image.shape
+
+    def _update(self):
+        """Update the underlying visual.
+        """
+        if self._need_display_update:
+            self._need_display_update = False
+
+            self.viewer.dims._child_layer_changed = True
+            self.viewer.dims._update()
+
+            self._node._need_colortransform_update = True
+            self._set_view_slice(self.viewer.dims.indices)
+
+        if self._need_visual_update:
+            self._need_visual_update = False
+            self._node.update()
+
+    def _refresh(self):
+        """Fully refresh the underlying visual.
+        """
+        self._need_display_update = True
+        self._update()
+
+    def _slice_image(self, indices):
+        """Determines the slice of image given the indices.
+
+        Parameters
+        ----------
+        indices : sequence of int or slice
+            Indices to slice with.
+        """
+        ndim = self.ndim
+        indices = list(indices)[:ndim]
+
+        for dim in range(len(indices)):
+            max_dim_index = self.image.shape[dim] - 1
+
+            try:
+                if indices[dim] > max_dim_index:
+                    indices[dim] = max_dim_index
+            except TypeError:
+                pass
+
+        return self.image[tuple(indices)]
+
+    def _set_view_slice(self, indices):
+        """Sets the view given the indices to slice with.
+
+        Parameters
+        ----------
+        indices : sequence of int or slice
+            Indices to slice with.
+        """
+        sliced_image = self._slice_image(indices)
+        self._node.set_data(sliced_image)
+
+        self._need_visual_update = True
+        self._update()
+
+    @property
+    def method(self):
```

Comment:
Why are you adding exposing this `method`? Is it something we will want to expose on our normal `image` layer too or is it something you only see needed for our `labels` layer. It is not quite clear to me what it is needed for

Reply:
This is code copy-pasted from the image layer. It's already there. No idea if we want to override it — my impression was no, but as I mentioned, this code was already there. =)

Reply:
ah - my mistake, i missed it last time i looked